mlir/include/mlir/IR/Remarks.h

```diff
@@ -60,7 +60,12 @@ struct RemarkOpts {
   StringRef categoryName;    // Category name (subject to regex filtering)
   StringRef subCategoryName; // Subcategory name
   StringRef functionName;    // Function name if available
-  RemarkOpts() = delete;
```

Comment:
I just wonder: Is it worthwhile making this class not default-constructible? Will there be any issue if we just implicitly declare all special member functions? @grypp

Reply:
Providing no-constructor should be fine I think.

Reply:
Works for me.

I believe this isn't available in c++17, but in c++20 it will allow construction like `{.remarkName = "name", ...}`, which IMO is more readable.